### PR TITLE
 feat: migrate dust apps v1 endpoints to vaults

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

This PR was previously rolled back as it caused assistants interactions to fail.
This was due to not passing the vault ID when calling registry apps (causing the global vault to be assumed). Registry apps are stored in a public vault (not in the global vault).

Moves the v1 endpoints for dust apps under `/vault`. Legacy endpoints are kept (they use the same handler as the new one).
Only legacy endpoints are documented for now. 

## Risk

Changing the whole permissions-checking logjc on dust apps public endpoint. 

## Deploy Plan

N/A

### CODE REVIEW

**Review by tdraier — APPROVED** [review]

File: front/pages/api/v1/w/[wId]/vaults/[vId]/apps/[aId]/runs/[runId]/index.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
+/* eslint jsdoc/no-missing-syntax: 0 */ //
+// Disabling jsdoc rule, as we're not yet documentating dust apps endpoints under vaults.
+// We still document the legacy endpoint, which does the same thing.
+// Note: for now, an API key only has access to the global vault.
+import type { RunType, WithAPIErrorResponse } from "@dust-tt/types";
```
Comment (flvndvd) [inline front/pages/api/v1/w/[wId]/vaults/[vId]/apps/[aId]/runs/[runId]/index.ts:5]: `nit`: IIRC we usually do this:

https://github.com/dust-tt/dust/blob/edab8258b0c22fa484f1b56c36937639196bd81b/front/pages/api/v1/w/%5BwId%5D/feature_flags.ts#L15-L18

File: front/pages/api/v1/w/[wId]/vaults/[vId]/apps/index.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  const owner = auth.workspace();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace you're trying to access was not found",
+      },
+    });
+  }
```
Comment (flvndvd) [inline front/pages/api/v1/w/[wId]/vaults/[vId]/apps/index.ts:32]: Not needed anymore with the new wrapper!

Comment (fontanierh) [reply front/pages/api/v1/w/[wId]/vaults/[vId]/apps/index.ts:32]: nice !

File: front/pages/api/v1/w/[wId]/vaults/[vId]/apps/[aId]/runs/[runId]/index.ts (hunk 1 above)
Comment (fontanierh) [reply front/pages/api/v1/w/[wId]/vaults/[vId]/apps/[aId]/runs/[runId]/index.ts:5]: TIL thx

**Pull request merged**